cmd/juju/caas/add.go

```diff
@@ -247,8 +254,13 @@ func addCredentialToLocal(cloudName string, newCredential cloud.Credential, cred
 	return nil
 }
 
+// This is exported to be patched in tests.
+var CurrentAccountDetails = func(c *AddCAASCommand) (*jujuclient.AccountDetails, error) {
+	return c.CurrentAccountDetails()
+}
+
 func (c *AddCAASCommand) addCredentialToController(apiClient CloudAPI, newCredential cloud.Credential, credentialName string) error {
-	currentAccountDetails, err := c.CurrentAccountDetails()
+	currentAccountDetails, err := CurrentAccountDetails(c)
```

Comment:
I don't know of a good reason why I pass in the cloudMetadataStore in the constructor but don't do the same for the FileCredentialStore. That seems like a straightforward improvement, and it'll make both addCredentialTo* functions methods.

As for CurrentAccountDetails, it's coming from ModelCommandBase, which AddCAASCommand embeds. It seems odd to me to include that in the constructor, although it might be less bad than this.